src/NuGet.Core/NuGet.Versioning/VersionComparer.cs

```diff
@@ -80,7 +92,7 @@ public static int Compare(SemanticVersion? version1, SemanticVersion? version2,
         /// <summary>
         /// Gives a hash code based on the normalized version string.
         /// </summary>
-        public int GetHashCode(SemanticVersion version)
+        public int GetHashCode(SemanticVersion? version)
```

Comment:
Same as my comment in VersionRangeComparer. I don't think that GetHashCode methods should allow nulls. When I added null annotations, I didn't mark this as nullable, even though the implementation allows it, because I'd rather API consumers to use better API designs.